Celeste.Mod.mm/MonoModRules.cs

```diff
@@ -2541,6 +2545,29 @@ bgt.un.s	ldloca.s
             // Target for if renamed is false
             cursor.MarkLabel(renamedTarget);
         }
+        
+        public static void PatchMoveBlockController(MethodDefinition method, CustomAttribute attrib) {
+            FieldDefinition f_this = null;
+            MethodReference m_Platform_DisableStaticMovers = MonoModRule.Modder.Module.GetType("Celeste.Platform").FindMethod("System.Void DisableStaticMovers()");
+
+            // The routine is stored in a compiler-generated method.
+            foreach (TypeDefinition nest in method.DeclaringType.NestedTypes) {
+                if (!nest.Name.StartsWith("<" + method.Name + ">d__"))
+                    continue;
+                method = nest.FindMethod("System.Boolean MoveNext()") ?? method;
+                break;
+            }
+
+            // Goal is to flip these two commands. We move to before the Move call, emit the Disable call, then remove the original Disable call.
+            // this.MoveStaticMovers(this.startPosition - this.Position);
+            // this.DisableStaticMovers();
+            ILCursor cursor = new ILCursor(new ILContext(method));
+            cursor.GotoNext(MoveType.After, instr => instr.OpCode == OpCodes.Blt);
```

Comment:
It would be preferable to target a more specific instruction, such as the `MoveStaticMovers`, instead of a `blt`. The `DisableStaticMovers` call could even be inserted immediately before the call to `MoveStaticMovers`, which still decompiles fairly well in ILSpy:
```cs
Vector2 amount = moveBlock.startPosition - moveBlock.Position;
moveBlock.DisableStaticMovers();
moveBlock.MoveStaticMovers(amount);
```
Alternatively jumping to `MoveStaticMovers` and then moving backwards to the `blt` (or two `ldloc_1`s) would be an equally viable solution